src/huggingface_hub/utils/_errors.py

```diff
@@ -57,6 +57,28 @@ def __init__(self, message, response):
         super().__init__(message, response=response)
 
 
+class LocalEntryNotFoundError(EntryNotFoundError, FileNotFoundError, ValueError):
+    """
+    Raised when trying to access a file that is not on the disk when network is
+    disabled. The entry may exist on the Hub.
+
+    Note: `ValueError` type is to ensure backward compatibility.
+    Note: `LocalEntryNotFoundError` derives from `HTTPError` because of `EntryNotFoundError`
+          even though it is not network related.
```

Comment:
It is a tiny bit in the way you raise it: it will be raised when `hf.co` is down and the file is not in the cache for instance.

Reply:
True, addressed in https://github.com/huggingface/huggingface_hub/pull/985/commits/83efe99c0a1c89edf6953e7b2475428abfae28ba.